File: Stack/1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int l = 0;

        int currSum = 0;
        int maxOnes = 0;

        for(int r=0;r<nums.length;r++) {
            currSum += nums[r];
            if((r-l+1)-currSum > 1) {
                currSum-=nums[l];
                l++;
            }

            maxOnes = Math.max(maxOnes, currSum);
        }
        if(maxOnes == nums.length) {
            return maxOnes-1;
        }
        return maxOnes;
    }
}

File: Array/1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        l = 0
        max_cons = 0
        count = 0
        
        for r in range(len(nums)):
            if nums[r] == 1:
                count+=1
            while l<len(nums) and (r-l+1)-count > 1:
                if nums[l] == 1:
                    count-=1
                l+=1
            max_cons = max(max_cons, count)
        
        return max_cons if max_cons <len(nums) else max_cons-1